functions/archiveEvents/source.js

```diff
@@ -129,32 +232,41 @@ dataDogEvent = function(title,message,tag){
   })
   .then(response => {
     const json_body = EJSON.parse(response.body.text());
-    return json_body;
+    if (typeof(json_body.status) == 'string' && json_body.status == "ok")
+    	return json_body;
+    Promise.reject(`Datadog: ${json_body.errors}`);
+  })
+  .catch(err => {
+	return Promise.reject(`DataDog: ${err}`);
   });
 };
 
 getMyOrg = function() {
   
   const url = `https://cloud.mongodb.com/api/atlas/v1.0/orgs`; 
-  return context.functions.execute('cloudAuthenticatedRequest', url, prefix)
+  return cloudAuthenticatedRequest( url, prefix)
     .then(response => {
       if (response.totalCount == 1) 
         return response.results[0].links[0].href.toString();
     })
-    .catch(err => console.error(`Failed to get Root Org: ${err}`));
+    .catch(err => {
+    	  return Promise.reject(`getMyOrg: Error getting Org info: \n-->${err}`);
+    });
 };
 
 getMyProjects = function() {
   content = [];
   const url = `https://cloud.mongodb.com/api/atlas/v1.0/groups`; 
-  return context.functions.execute('cloudAuthenticatedRequest', url, prefix)
+  return cloudAuthenticatedRequest( url, prefix)
     .then(response => {
       for(var i=0;i<response.results.length;i++){
         content.push(response.results[i].links[0].href.toString());
       }
       return(content);
     })
-    .catch(err => console.error(`Failed to get my projects: ${err}`));
+    .catch(err => {
+  	  return Promise.reject(`getMyProjects: Error getting Project info: \n-->${err}`);
+  });
```

Comment:
Everything seems correct to me, except the indentation of this line, which doesn't matter. All errors should bubble up to the top now.